client/lib/wp/sync-handler/index.js

```diff
@@ -121,26 +163,31 @@ export class SyncHandler {
 			 * WP.com server-side response.
 			 *
 			 * @param {Object} serverResponse - server response object
+			 * @returns {Promise} - promise that resolves with server response object
 			 */
 			const cacheResponse = serverResponse => {
-				// get response object without _headers property
-				let responseWithoutHeaders = Object.assign( {}, serverResponse );
-				delete responseWithoutHeaders._headers;
-
-				let storingData = {
-					__sync: {
-						key,
-						synced: Date.now(),
-						syncing: false
-					},
-					body: responseWithoutHeaders,
-					params: reqParams
-				};
-
-				// add/override gotten data from server-side
-				this.storeRecord( key, storingData ).catch( err => {
-					// @TODO error handling
-					warn( err );
+				return new Promise( ( resolve ) => {
```

Comment:
Sorry, the comment above maybe doesn't make sense.
I think you're trying to pass the server-side response to the next `then()`, call which means to `checkPagination()` function. If this is the purpose you could just return the response:

```es6
const cacheResponse = serverResponse => {
  // ...

  return serverResponse;
}
```